core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -855,6 +856,23 @@ sealed abstract class ByteVector extends BitwiseOperations[ByteVector, Long] wit
     bits.toLong(signed, ordering)
 
   /**
+    * Converts the contents of this vector to a UUID.
+    *
+    * @throws IllegalArgumentException if size is not exactly 8.
+    */
+  final def toUUID: UUID = {
+    // Sanity check
+    if (size != 16) {
```

Comment:
I thought this was better than > 16 for simple safety reasons. (Also: Principle of Least Surprise)